### ISSUE TITLE
918838 - [Bug] Tarpit Trap Multiple bugs and issues

### ISSUE THREAD

**Issue by PvtVoid** [Talishar/Talishar#378]
### Following this tweet here are the issues with the card (https://twitter.com/KodachiOne/status/1635331699377848322)

1. I block an attack with go again so Tarpit Trap should be active and prevent the next on-hit attack. It doesn't affect only the attack it's defending. "the next time an attack action card hits this turn, effects don't trigger."

![image](https://user-images.githubusercontent.com/26232486/224791263-87b7d8b2-8fe5-43b8-a325-2b98f6c79857.png)
![image](https://user-images.githubusercontent.com/26232486/224791499-b425b326-02ac-45f3-9c89-fc526ba0dd62.png)

2. After it hit everything like Katsu and Mask of Momentum should have triggered anyway but be prevented. So they shouldn't be active anymore and cannot trigger a 2nd time.

```
6.6.5f If the triggered effect is triggered, but there is an effect that prevents it from triggering, the triggered layer is not created and will not be added to the stack. However, the triggering still counts towards the limit of the effect (if any).

So it should work out that, pending release notes and CR update saying otherwise, Katsu doesn't trigger on this attack or at all this turn and mask doesn't trigger on this attack or at all this turn
```

3. It is still a hit but shows as blocked: 
![image](https://user-images.githubusercontent.com/26232486/224792370-16b44047-febd-482a-8db4-f67188311c75.png)
![image](https://user-images.githubusercontent.com/26232486/224792818-3c41328f-6bdd-4fd7-836c-a4aa57790735.png)

Plus Mask + Kastu as still active:
![image](https://user-images.githubusercontent.com/26232486/224793027-54d940d2-0897-4694-8849-44bf182c188d.png)

**Comment by PvtVoid** [comment]
If you have questions about interactions or rulings, please contact the judge community for clarification and it'll be a pleasure to help 🐝

**Comment by OotTheMonk** [comment]
The first issue is fixed with revision c3c4f830, I think the third one should be fixed incidentally too due to the difference in how the hit effect is prevented. Writing the second issue as a separate issue.
